tests/samples/schema.org/product-ref.json

```diff
@@ -0,0 +1,69 @@
+[
+    {
+        "type": "http://schema.org/Product",
+        "properties": {
+            "referenced_product": {
+                "type": "http://schema.org/Product",
+                "properties": {
+                    "name": "REFERENCED PRODUCT",
+                    "image": "img-ref.jpg"
+                }
+            },
+            "prop3": "REFERENCED TO INCLUDE PROPERTIES AND ALSO INDIVIDUAL PRODUCT",
+            "image": [
+                "img-2.jpg",
+                "anvil_executive.jpg"
+            ],
+            "related_products": [
+                {
+                    "type": "http://schema.org/Product",
+                    "properties": {
+                        "name": "REL PROD 1",
+                        "image": "rel-prod-1.jpg"
+                    }
+                },
+                {
+                    "type": "http://schema.org/Product",
+                    "properties": {
+                        "name": "REL PROD 2",
+                        "image": "rel-prod-2.jpg"
+                    }
+                }
+            ],
+            "brand": "ACME",
+            "name": "Executive Anvil",
+            "description": "Sleeker than ACME's Classic Anvil, the\n      Executive Anvil is perfect for the business traveler\n      looking for something to drop from a height.",
+            "mpn": "925872",
+            "aggregateRating": {
+                "type": "http://schema.org/AggregateRating",
+                "properties": {
+                    "ratingValue": "4.4",
+                    "reviewCount": "89"
+                }
+            },
+            "offers": {
+                "type": "http://schema.org/Offer",
+                "properties": {
+                    "priceCurrency": "USD",
+                    "price": "119.99",
+                    "priceValidUntil": "2020-11-05",
+                    "seller": {
+                        "type": "http://schema.org/Organization",
+                        "properties": {
+                            "name": "Executive Objects"
+                        }
+                    },
+                    "itemCondition": "http://schema.org/UsedCondition",
+                    "availability": "http://schema.org/InStock"
+                }
+            }
+        }
+    },
+    {
+        "type": "http://schema.org/Product",
+        "properties": {
+            "prop3": "REFERENCED TO INCLUDE PROPERTIES AND ALSO INDIVIDUAL PRODUCT",
```

Comment:
Sorry maybe I'm reading the schema incorrectly, but why do we have a separate product with its properties duplicated also in the product above?

Reply:
The product with id `more-properties` does not have attribute `itemprop`. Because of that, when referenced only inner attributes are included. As the product has not been extracted when referencing, it is finally extracted as an individual Product.

Regarding if that should be behaviour or not, I'm not completelly sure. This is the standard definition of `itemref`https://www.w3.org/TR/microdata/#names:-the-itemprop-attribute but is not very complete. 

There is an algorithm definition here: https://www.w3.org/TR/microdata/#associating-names-with-items but I didn't go so deep for this patch as it seems it is not the same approach followed by `extruct`.

I have just tested this html in Google Dev Tools and it also extracts the product isolated, but it is not including `prop3` into the first product.

Reply:
I see, thanks for checking this @ivanprado . I think the current approach makes sense for ``extruct``.